src/module-elasticsuite-catalog-rule/Model/Rule/Condition/Product/SpecialAttribute/ProductDate.php

```diff
@@ -0,0 +1,194 @@
+use Smile\ElasticsuiteCatalogRule\Model\Rule\Condition\Product as ProductCondition;
+use Smile\ElasticsuiteCore\Search\Request\Query\QueryFactory;
+use Smile\ElasticsuiteCore\Search\Request\QueryInterface;
+
+/**
+ * Special product dates attribute class.
+ *
+ * @category Smile
+ * @package  Smile\ElasticsuiteCatalogRule
+ * @author   Vadym Honcharuk <vahonc@smile.fr>
+ */
+class ProductDate implements SpecialAttributeInterface
+{
+    /**
+     * @var string
+     */
+    private $attributeCode;
+
+    /**
+     * @var string
+     */
+    private $attributeLabel;
+
+    /**
+     * @var QueryFactory
+     */
+    protected $queryFactory;
+
+    /**
+     * ProductDate constructor.
+     *
+     * @param string       $attributeCode  Attribute Code.
+     * @param string       $attributeLabel Attribute Label.
+     * @param QueryFactory $queryFactory   Query factory.
+     */
+    public function __construct(
+        string $attributeCode,
+        string $attributeLabel,
+        QueryFactory $queryFactory
+    ) {
+        $this->attributeCode  = $attributeCode;
+        $this->attributeLabel = $attributeLabel;
+        $this->queryFactory   = $queryFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAttributeCode(): string
+    {
+        return $this->attributeCode;
+    }
+
+    /**
+     * {@inheritdoc}
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function getSearchQuery(ProductCondition $condition)
+    {
+        // Retrieve the operator and days from the condition.
+        $operator = $condition->getOperator();
+        $value = (int) $condition->getValue();
+
+        // Calculate the reference date based on the number of days.
+        $dateReference = new \DateTime();
+        $dateReference->setTime(0, 0, 0); // Set the time to the start of the day.
+        $dateReference->modify('-' . $value . ' days');
+
+        // Prepare the bounds based on the operator.
+        $bounds = [];
+        switch ($operator) {
+            case '==':
+            case '!=':
+                $bounds['gte'] = $dateReference->format('Y-m-d');
+                $bounds['lt'] = $dateReference->modify('+1 day')->format('Y-m-d');
+                break;
+            case '>':
+                $bounds['lt'] = $dateReference->format('Y-m-d');
+                break;
+            case '<':
+                $bounds['gt'] = $dateReference->format('Y-m-d');
+                break;
+            case '>=':
+                $bounds['gte'] = $dateReference->format('Y-m-d');
+                break;
+            case '<=':
+                $bounds['lte'] = $dateReference->format('Y-m-d');
+                break;
+            default:
+                throw new \InvalidArgumentException('Invalid operator');
+        }
+
+        $fieldName = ($condition->getAttribute() === 'is_created_within_last_x_days') ? 'created_at' : 'updated_at';
```

Comment:
fetch $fieldName from the DI, it will be cleaner.

Reply:
Done.

Reply:
cool ! gonna test this asap, but the code is clean, good job !

Reply:
@vahonc `lte` is not working, I just tested, and it is pulling data that is 3 months old

Reply:
@IbrahimS2, are you sure there are no incorrect date settings? Can you add a real example (perhaps in a related issue [#3260](https://github.com/Smile-SA/elasticsuite/issues/3260))?

Reply:
@vahonc I am sure, but I think it is related to #3274, meaning I did not have this PR in my local; are there any other PRs relevant to this PR dependent upon?

![image](https://github.com/Smile-SA/elasticsuite/assets/19901858/a9995cf2-7797-4758-8666-fa9bae4c6700)

Reply:
@IbrahimS2 you were right, I fixed the issue in 81413be34e079d37b2bf7673b043dfd079ccbf2d and published a release. Enjoy !